pkg/jiraservicedesk/client/client.go

```diff
@@ -24,7 +24,7 @@ type Client interface {
 	UpdateProject(updatedProject Project, id string) error
 	ProjectEqual(oldProject Project, newProject Project) bool
 	GetProjectForUpdateRequest(existingProject Project, newProject *jiraservicedeskv1alpha1.Project) Project
-	UpdateProjectAccessPermissions(status bool, key string) error
+	ModifyProjectAccessPermissions(status bool, key string) error
```

Comment:
This should be `UpdateProjectAccessPermissions`. Btw my point was that when a project is updated, we need to check for this flag and send an API call if this value is changed